resources/js/register-components.tsx

```diff
@@ -106,17 +107,24 @@ core.reactTurbolinks.register('ranking-variant-filter', () => (
   <RankingVariantFilter {...parseJson('json-variant-filter')} />
 ));
 
+core.reactTurbolinks.register('store-supporter-tag', (container) => {
+  const productId = getInt(container.dataset.productId);
+  if (productId == null) {
+    throw new Error('missing productId');
+  }
+
+  const maxMessageLength = getInt(container.dataset.maxMessageLength) ?? 0;
```

Comment:
weird missing attribute handling with one throwing and one having nonsense default

(and similar to quantity, having product id inside the component just adds complexity compared to just having it directly on plain html)